common/base/include/claragenomics/utils/device_buffer.hpp

```diff
@@ -52,32 +54,41 @@ class buffer
     /// \brief This constructor creates \p buffer with the given size.
     /// \param n The number of elements to create
     /// \param allocator The allocator to use by this buffer.
-    /// \param stream The CUDA stream to be associated with this allocation. Default is stream 0.
+    /// \param streams CUDA streams to be associated with this buffer. Freeing the memory (either through destructor, free() or clear_and_resize()) calls cudaStreamSynchronize() on all those streams. If no stream is specified default stream (0) is used.
```

Comment:
Please don't document the "calls cudaStreamSynchronize" - we might want to change that at some point.
Instead you could write "the device memory provided by this buffer is guaranteed to live until all operations on the associated stream have completed."

Reply:
done